data-prepper-plugins/sqs-source/src/main/java/org/opensearch/dataprepper/plugins/source/sqs/SqsWorker.java

```diff
@@ -172,9 +172,13 @@ private List<DeleteMessageBatchRequestEntry> processSqsEvents(final List<Message
         }
 
         for (Message message : messages) {
-            Duration duration = Duration.between(Instant.ofEpochMilli(Long.parseLong(message.attributes().get(MessageSystemAttributeName.SENT_TIMESTAMP))), Instant.now());
-            sqsMessageDelayTimer.record(duration.isNegative() ? Duration.ZERO : duration); // Negative durations can occur if messages are processed immediately
-
+            final int receiveCount = Integer.parseInt(message.attributes().get(MessageSystemAttributeName.APPROXIMATE_RECEIVE_COUNT));
```

Comment:
If this attribute doesn't exist for some reason, will this default to 0, or will Integer.parseInt(null) throw an exception?

Reply:
both of these system attributes will always exist since we make this receiveMessageRequest call:

       ReceiveMessageRequest.Builder requestBuilder = ReceiveMessageRequest.builder()
                .queueUrl(queueUrl)
                .attributeNamesWithStrings("All")
                .messageAttributeNames("All");